vehicle/carwings.go

```diff
@@ -57,177 +39,52 @@ func NewCarWingsFromConfig(other map[string]interface{}) (api.Vehicle, error) {
 
 	log := util.NewLogger("carwings").Redact(cc.User, cc.Password, cc.VIN)
 
-	// http client with high dial/handshake timeout
-	transport := request.NewTripper(log, &http.Transport{
-		Proxy: http.ProxyFromEnvironment, // default
-		DialContext: (&net.Dialer{
-			Timeout:   carwingsRequestTimeout,
-			KeepAlive: 30 * time.Second, // default
-		}).DialContext,
-		TLSHandshakeTimeout:   carwingsRequestTimeout,
-		ForceAttemptHTTP2:     true,             // default
-		MaxIdleConns:          100,              // default
-		IdleConnTimeout:       90 * time.Second, // default
-		ExpectContinueTimeout: 1 * time.Second,  // default
-	})
-
-	carwings.Client = &http.Client{
-		Timeout:   carwingsRequestTimeout,
-		Transport: transport,
+	v := &CarWings{
+		embed: &cc.embed,
 	}
 
-	v := &CarWings{
-		embed:    &cc.embed,
-		user:     cc.User,
-		password: cc.Password,
-		session: &carwings.Session{
-			Region: cc.Region,
-			VIN:    cc.VIN,
-		},
+	identity := carwings.NewIdentity(log)
+	login, err := identity.Login(cc.User, cc.Password)
+	if err != nil {
+		return v, fmt.Errorf("login failed: %w", err)
 	}
 
-	// initial connect
-	if err := v.session.Connect(v.user, v.password); err != nil {
-		return nil, fmt.Errorf("login failed: %w", err)
+	api := carwings.NewAPI(log, login.VehicleInfo.CustomSessionID, login.CustomerInfo.Timezone)
+	err = CheckVIN(login, cc.VIN)
```

Comment:
This is different from all other providers. Would be nice to align the implementations.